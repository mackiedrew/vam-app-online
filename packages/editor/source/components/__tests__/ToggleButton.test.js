import Subject from "../ToggleButton";

describe("<ToggleButton />", () => {
  describe("renders without crashing", () => {
    it("while off", () => {
      shallow(<Subject />);
    });
    it("while on", () => {
      shallow(<Subject on />);
    });
  });

  describe("renders correctly", () => {
    it("while off", () => {
      const tree = renderer.create(<Subject />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("while on", () => {
      const tree = renderer.create(<Subject on />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("button calls appropriate function", () => {
    it("while off, calls on function", () => {
      const onFunctionMock = sinon.spy();
      const offFunctionMock = sinon.spy();
      const subject = shallow(
        <Subject offFunction={offFunctionMock} onFunction={onFunctionMock} />
      );
      subject.find("button.toggle-button").simulate("click");
      expect(onFunctionMock.called).toBe(false);
      expect(offFunctionMock.called).toBe(true);
    });
    it("while on, calls on function", () => {
      const onFunctionMock = sinon.spy();
      const offFunctionMock = sinon.spy();
      const subject = shallow(
        <Subject offFunction={offFunctionMock} onFunction={onFunctionMock} on />
      );
      subject.find("button.toggle-button").simulate("click");
      expect(onFunctionMock.called).toBe(true);
      expect(offFunctionMock.called).toBe(false);
    });
  });
});
