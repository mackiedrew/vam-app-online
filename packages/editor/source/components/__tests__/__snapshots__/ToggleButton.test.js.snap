// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> renders correctly while off 1`] = `
<button
  className="toggle-button"
  onClick={undefined}
/>
`;

exports[`<ToggleButton /> renders correctly while on 1`] = `
<button
  className="toggle-button"
  onClick={undefined}
/>
`;
